src/benchmark_register.cc

```diff
@@ -62,6 +62,23 @@ namespace internal {
 //=============================================================================//
 //                         BenchmarkFamilies
 //=============================================================================//
+int64_t GetBenchmarkUniqueId() {
```

Comment:
Why is the base name and overall benchmark name not enough for this?

Reply:
The base name can be similar between several benchmarks. See the example posted on the last PR, where a NOP function with only slight variations in repetitions, threads, etc. were applied: https://github.com/google/benchmark/pull/567#issuecomment-386820365